src/extensions/service-managers/twilio-account-switching/index.js

```diff
@@ -0,0 +1,228 @@
+import { accessRequired } from "../../../server/api/errors";
+import { getFeatures } from "../../../server/api/lib/config";
+import { cacheableData, r } from "../../../server/models";
+import { getSecret, convertSecret } from "../../secret-manager";
+
+let orgChanges, orgFeatures;
+let version = 1;
+
+export const name = "twilio-account-switching";
+
+export const metadata = () => ({
+  displayName: "Twilio Accounts",
+  description: "Add, delete, and edit Twilio accounts",
+  canSpendMoney: false,
+  moneySpendingOperations: ["onCampaignStart"],
+  supportsOrgConfig: true,
+  supportsCampaignConfig: true
+});
+
+export async function onMessageSend({
+  message,
+  contact,
+  organization,
+  campaign
+}) {
+  const campaignTwilioAccount = _getCampaignTwilioAccount(
+    campaign,
+    ({ accountSid, authToken, id, messageServiceSids }) => ({
```

Comment:
That pattern is the "Object Destructuring and Property Shorthand" (https://stackoverflow.com/questions/17781472/how-to-get-a-subset-of-a-javascript-objects-properties)

`({ accountSid, authToken, id, messageServiceSids }) => ({accountSid, authToken, id, messageServiceSids})` gets passed into the `_getCampaignTwilioAccount` function as the `keys` parameter

The `_getCampaignTwilioAccount` function is defined as:
https://github.com/MoveOnOrg/Spoke/blob/fad5492179fb35d3378852bad0d1bf1d343c0980/src/extensions/service-managers/twilio-account-switching/index.js#L196-L217

`keys` defines the keys that the function will return from a Twilio account object. In this case, it will return the accountSid, authToken, id, and messageServiceSids keys from the Twilio account object. In some cases, the `_getCampaignTwilioAccount` function may be used as a helper function to return data to the front-end, such as here in `getCampaignData`: https://github.com/MoveOnOrg/Spoke/blob/fad5492179fb35d3378852bad0d1bf1d343c0980/src/extensions/service-managers/twilio-account-switching/index.js#L72-L76, so the purpose of using the "Object Destructuring and Property Shorthand" pattern is to return only the required keys from the Twilio account object, instead of all the keys

Reply:
ahhhh yes that makes a lot of sense. Thank you for this explanation!